src/reducers/paths.reducer.js

```diff
@@ -46,32 +47,42 @@ const initialState = {
 
 export default (state: State = initialState, action: Action) => {
   switch (action.type) {
-    case ADD_PROJECT:
+    case ADD_PROJECT: {
+      const { project } = action;
+
+      const projectNameSlug = getProjectNameSlug(project.guppy.name);
+
+      return produce(state, draftState => {
+        draftState.byId[project.guppy.id] = formatProjectPath(
+          state.homePath,
+          projectNameSlug
+        );
+      });
+    }
+
     case IMPORT_EXISTING_PROJECT_FINISH: {
       const { projectPath, project } = action;
+
       return produce(state, draftState => {
-        draftState.byId[project.guppy.id] =
-          projectPath || formatProjectPath(state.homePath, project.guppy.id);
+        draftState.byId[project.guppy.id] = projectPath;
       });
     }
 
-    case CHANGE_PROJECT_HOME_PATH: {
-      const { homePath } = action;
+    case SAVE_PROJECT_SETTINGS_FINISH: {
+      const { project, projectPath } = action;
+
       return produce(state, draftState => {
-        draftState.homePath = homePath;
+        draftState.byId[project.guppy.id] = projectPath;
       });
     }
-    case SAVE_PROJECT_SETTINGS_FINISH: {
-      const { project, projectPath, oldProjectId } = action;
 
+    case CHANGE_PROJECT_HOME_PATH: {
+      const { homePath } = action;
       return produce(state, draftState => {
-        // remove oldId if id changed & add new path
-        if (oldProjectId !== project.guppy.id) {
-          delete draftState.byId[oldProjectId];
-          draftState.byId[project.guppy.id] = projectPath;
-        }
```

Comment:
`oldProjectId` was always undefined, so this condition was always true. Happily, I think that works; even if the user hasn't changed the path, there's no harm in setting it to itself.